Test/dafny0/NameclashesCompile.dfy

```diff
@@ -0,0 +1,44 @@
+// RUN: %dafny /compile:3 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+method Main() {
+  var r0, r1;
+  
+  var d := Ctor(15);
+  r0 := d.ToString();
+  r1 := d.GetHashCode();
+  print r0, " ", r1, "\n";
+  
+  var c := CoCtor(15);
+  r0 := c.ToString();
+  r1 := c.GetHashCode();
+  print c.c, " ", c.d, " ", c.Computer(), " ";
+  print r0, " ", r1, "\n";
+  
+  var n: NameclashNew := 3;
+  r0 := n.ToString();
+  r1 := n.GetHashCode();
+  print r0, " ", r1, "\n";
+}
+
+datatype Nameclash = Ctor(x: int)
```

Comment:
I don't understand the example.  Where is the name clash?

Reply:
There are two sets of name clashes. One is the names `ToString` and `GetHashCode`, which exist in .NET as overridable methods for all classes. The Dafny compiler for C# previously passed these names through, which causes the C# compiler to complain that there's a method that shadows a virtual method (which would commonly be a mistake). The other set of name clashes comes from the fields `c` and `d` that the Dafny-to-C# compiler generates for `codatatype`s.

This file thus contains regression tests for these previous bugs.